### PR TITLE
[YAML] Allow hex: prefix for octet strings

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

Octet strings can not be specified using the `hex:` notation.

#### Change overview
 * Supports `hex:` prefix for octet strings

#### Testing
A test has been added to TestCluster.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by tcarmelveilleux — APPROVED** [review]

File: src/app/zap-templates/common/ClusterTestGeneration.js [hunk 1]
```diff
@@ -826,6 +826,28 @@ function isLiteralNull(value, options)
...
 }
 
+function isHexString(value)
+{
+  return value && value.startsWith('hex:');
+}
+
+function octetStringFromHexString(value)
+{
+  const hexString = value.substring(4);
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/common/ClusterTestGeneration.js:836]: Might be less fragile to have a `kHexPrefix` string, and use `kHexPrefix.length` here and in the next function.

File: src/app/zap-templates/common/ClusterTestGeneration.js [hunk 2]
```diff
@@ -826,6 +826,28 @@ function isLiteralNull(value, options)
...
+
+  const bytes = [];
+  for (let i = 0; i < hexString.length; i += 2) {
+    bytes.push(hexString.substring(i, i + 2));
+  }
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/common/ClusterTestGeneration.js:841]: Not sure whether this is clearer than:
```suggestion
  const bytes = hexString.match(/(..)/g);
```
or not... But also, can/should we throw an error if the number of chars is odd or some char is not in `[0-9a-fA-F]`?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/zap-templates/common/ClusterTestGeneration.js (hunk 2 above)
Comment (vivien-apple) [reply src/app/zap-templates/common/ClusterTestGeneration.js]: I have updated the code with your suggestion and I have also added errors if the number of characters is odd and if there is any invalid hexadecimal characters.

File: src/app/tests/suites/TestCluster.yaml [hunk 3]
```diff
@@ -727,6 +727,18 @@ tests:
...
+      command: "writeAttribute"
+      attribute: "octet_string"
+      arguments:
+          value: "hex:000102030405"
+
+    - label: "Read attribute OCTET_STRING with hex: format"
+      command: "readAttribute"
+      attribute: "octet_string"
+      response:
+          value: "hex:000102030405"
```
[the remark this answers is not in the document]
Comment (vivien-apple) [reply src/app/tests/suites/TestCluster.yaml:740]: That would makes sense but it requires a much bigger amount of changes in the handle due to the way "value" is handled everywhere.
So maybe in a followup when I have more time (not tomorrow :/ )

**Pull request merged**